subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/JavaGradlePluginPlugin.java

```diff
@@ -152,7 +152,17 @@ private void registerPlugins(Project project, GradlePluginDevelopmentExtension e
 
     private void applyDependencies(Project project) {
         DependencyHandler dependencies = project.getDependencies();
-        dependencies.add(API_CONFIGURATION, dependencies.gradleApi());
+        String apiVersion = System.getProperty("org.gradle.api.version");
```

Comment:
We query this property in six different places by name. Instead it would be good to have a shared mechanism to expose this as an `Optional<VersionNumber>` instead. An `@Inject`-able service would be nice.

Reply:
I extracted the common logic to a single class. I didn't make it a service, yet, since I am not sure what scope this would need to go.